denops/glance/main.ts

```diff
@@ -20,37 +35,63 @@ export async function main(denops: Denops) {
     const content = lines.join("\n");
     const document = await renderer.render(content);
     const pos = await fn.getpos(denops, ".");
     server.send("update", { document, line: pos[1] });
   }
-  const port = (await g.get(denops, "glance#server_port", 8765))!;
-  const plugins = (await g.get(denops, "glance#markdown_plugins", []))!;
-  const html = (await g.get(denops, "glance#markdown_html", false))!;
-  const breaks = (await g.get(denops, "glance#markdown_breaks", false))!;
-  const linkify = (await g.get(denops, "glance#markdown_linkify", false))!;
-  const defaultStylesheet = `#root {margin: 50px auto; width: min(700px, 90%);}`;
-  const stylesheet = (await g.get(denops, "glance#stylesheet", defaultStylesheet))!;
-  const defaultConfigPath = new URL("./config.ts", import.meta.url).toString();
-  const configPath = (await g.get(denops, "glance#config", defaultConfigPath))!;
-  const { createMarkdownRenderer } = await import(configPath);
-  const renderer = new MarkdownRenderer();
-  await renderer.initialize({ html, breaks, linkify, plugins, createMarkdownRenderer });
-  const server = new Server({ onOpen: update, readFile, stylesheet });
+
+  async function ensureOptions() {
+    if (options) return options;
+    const defaultStylesheet = `#root {margin: 50px auto; width: min(700px, 90%);}`;
+    const defaultConfigPath = new URL("./config.ts", import.meta.url).toString();
+    options = {
+      port: await g.get(denops, "glance#server_port", 8765),
+      plugins: await g.get(denops, "glance#markdown_plugins", []),
+      html: await g.get(denops, "glance#markdown_html", false),
+      breaks: await g.get(denops, "glance#markdown_breaks", false),
+      linkify: await g.get(denops, "glance#markdown_linkify", false),
+      defaultStylesheet,
+      stylesheet: await g.get(denops, "glance#stylesheet", defaultStylesheet),
+      defaultConfigPath,
+      configPath: await g.get(denops, "glance#config", defaultConfigPath),
+    };
+    return options;
+  }
+
+  async function ensureRenderer() {
+    if (renderer) return renderer;
+    options = await ensureOptions();
+    const { createMarkdownRenderer } = await import(options.configPath);
+    renderer = new MarkdownRenderer();
+    const { html, breaks, linkify, plugins } = options;
+    await renderer.initialize({ html, breaks, linkify, plugins, createMarkdownRenderer });
+    return renderer;
+  }
+
+  async function ensureServer() {
+    if (server) return server;
+    options = await ensureOptions();
+    server = new Server({ onOpen: update, readFile, stylesheet: options.stylesheet });
+    return server;
+  }
 
   denops.dispatcher = {
     update() {
       update();
       return Promise.resolve();
     },
-    listen() {
-      server.listen({ port });
+    async listen() {
+      options = await ensureOptions();
+      server = await ensureServer();
+      server.listen({ port: options.port });
       return Promise.resolve();
```

Comment:
According to the specification of `async` -- `await`, the last line `return Promise.reoslve()` is redundant in `async funciton`. Could you remove this line?

Reply:
Sure. Fixing.